ratik/src/intTest/java/org/opencadc/ratik/InventoryValidatorTest.java

```diff
@@ -0,0 +1,855 @@
+     * evidence: ?
+     * action: assume explanation3
+     *
+     * explanations 6 and 7 are covered by explanation 3
+     */
+
+    /**
+     * discrepancy: artifact.uri in both && artifact.id mismatch
+     *
+     * explanation1: same ID collision due to race condition that metadata-sync has to handle
+     * evidence: no more evidence needed
+     * action: pick winner, create DeletedArtifactEvent for loser, delete loser if it is in L
+     *
+     * case 1:
+     * local Artifact contentModifiedDate before remote Artifact contentModifiedDate, delete local Artifact.
+     * before: Artifact.uri in L && R, L Artifact.id != R Artifact.id,
+     *         L Artifact.contentModifiedDate < R Artifact.contentModifiedDate
+     * after:  Artifact not in L, DeletedArtifactEvent in L
+     *
+     * case 2:
+     * local Artifact contentModifiedDate after remote Artifact contentModifiedDate, no action
+     * before: Artifact.uri in L && R, L Artifact.id != R Artifact.id,
+     *         L Artifact.contentModifiedDate >= R Artifact.contentModifiedDate
+     * after:  Artifact in L
+     */
+    @Test
+    public void artifactUriCollision_LocalIsStorage() throws Exception {
+        artifactUriCollision(false);
+    }
+
+    @Test
+    public void artifactUriCollision_LocalIsGlobal() throws Exception {
+        artifactUriCollision(true);
+    }
+
+    public void artifactUriCollision(boolean trackSiteLocations) throws Exception {
+        // case 1: local contentModifiedDate before remote contentModifiedDate
+        URI contentCheckSum = TestUtil.getRandomMD5();
+        UUID localID = UUID.randomUUID();
+        UUID remoteID = UUID.randomUUID();
+        Calendar contentModifiedDate = Calendar.getInstance();
+
+        Artifact remoteArtifact = new Artifact(remoteID, URI.create("cadc:INTTEST/one.ext"), contentCheckSum,
+                                               contentModifiedDate.getTime(), 1024L);
+        this.remoteEnvironment.artifactDAO.put(remoteArtifact);
+
+        contentModifiedDate.roll(Calendar.YEAR, false);
+        Artifact localArtifact = new Artifact(localID, URI.create("cadc:INTTEST/one.ext"), contentCheckSum,
+                                              contentModifiedDate.getTime(), 1024L);
+        this.localEnvironment.artifactDAO.put(localArtifact);
+
+        try {
+            System.setProperty("user.home", TMP_DIR);
+            InventoryValidator testSubject = new InventoryValidator(this.localEnvironment.daoConfig, TestUtil.LUSKAN_URI,
+                                                                    new IncludeArtifacts(),null,
+                                                                    trackSiteLocations);
+            testSubject.run();
+        } finally {
+            System.setProperty("user.home", USER_HOME);
+        }
+
+        localArtifact = this.localEnvironment.artifactDAO.get(localArtifact.getID());
+        Assert.assertNull("local artifact found", localArtifact);
+
```

Comment:
The assert that localArtifact was deleted is part of the resolve (pick winner implied insert winning Artifact if it winner was in R). So this should also assert that remoteArtifact is now in localEnvironment.

I will update the doc.

Reply:
Fixed.